src/update_roster.R

```diff
@@ -1,160 +1,322 @@
-    dplyr::mutate(pfr_id = dplyr::if_else(is.na(pfr_id), pfr_id_joined, pfr_id)) |>
-    dplyr::select(-pfr_id_joined)
-  saveRDS(comb, "data/nflfastR-roster.rds")
-  readr::write_csv(comb, "data/nflfastR-roster.csv.gz")
+    roster <- roster |>
+      dplyr::mutate(headshot_url = gsub("\\{formatInstructions\\}", "f_auto,q_auto", headshot)) |>
+      dplyr::select(
+        season,
+        team = team_abbr,
+        position,
+        depth_chart_position,
+        jersey_number,
+        status = status_short_description,
+        full_name = display_name,
+        first_name,
+        last_name,
+        birth_date,
+        height,
+        weight,
+        college,
+        gsis_id,
+        espn_id,
+        sportradar_id,
+        yahoo_id,
+        rotowire_id,
+        pff_id,
+        pfr_id,
+        fantasy_data_id,
+        sleeper_id,
+        years_exp,
+        headshot_url,
+        dplyr::any_of(
+          c(
+            "ngs_position",
+            "esb_id",
+            "gsis_it_id",
+            "smart_id",
+            "entry_year",
+            "rookie_year"
+          )
+        )
+      ) |>
+      dplyr::mutate(
+        team = dplyr::case_when(team == "LAR" ~ "LA",
+                                team == "OAK" ~ "LV",
+                                TRUE ~ team),
+        height = stringr::str_remove_all(height, "\\\""),
+        height = stringr::str_replace_all(height, "'", "-"),
+        birth_date = lubridate::as_date(birth_date)
+      ) |>
+      dplyr::arrange(team, position)
 
-  # Safe RB & FB gsis IDs for usage in RACR computation
-  rb_ids <- comb |>
-    dplyr::filter(position %in% c("RB", "FB"), !is.na(gsis_id)) |>
-    dplyr::select(gsis_id) |>
-    dplyr::distinct() |>
-    dplyr::arrange(dplyr::desc(gsis_id))
+    cli::cli_alert_info("Save combined rosters...")
+    # nflversedata::nflverse_save(
+    #   data_frame = roster,
+    #   file_name =  glue::glue("roster_{season}"),
+    #   nflverse_type = "roster data",
+    #   release_tag = "rosters"
+    # )
 
-  qs::qsave(rb_ids, 'data/nflfastR-RB_ids.qs',
-            preset = "custom",
-            algorithm = "zstd_stream",
-            compress_level = 22,
-            shuffle_control = 15)
+    return(weekly_rosters)
+    # rm(list = ls())
+    cli::cli_alert_info("DONE!")
 
-  rm(list = ls())
-  cli::cli_alert_info("DONE!")
-}
-build_rosters()
-# update older seasons
-# we run this manually if necessary
-# purrr::walk(1999:(latest_season-1), function(i, comb){
-#   roster <- comb |> dplyr::filter(season == i)
-#   saveRDS(roster, glue::glue("data/seasons/roster_{unique(roster$season)}.rds"))
-#   readr::write_csv(roster, glue::glue("data/seasons/roster_{unique(roster$season)}.csv"))
-# }, comb = comb)
+  }
 
+# build ALL rosters
+# purrr::walk(1920:2022,build_rosters)
 
+# build most recent roster
+build_rosters(nflreadr:::most_recent_season())
```

Comment:
I think it's better to remove the argument here, i.e. `build_rosters()` as the default is correct. This is missing `roster = TRUE` inside `most_recent_season` imo